### ISSUE TITLE
Wrong line number references after ```yaml code blocks with triple dashes

### ISSUE THREAD

**Issue by villavic** [markdownlint/markdownlint#508]
## Description

Code blocks that "_look like_" YAML `frontmatter` cause line counts to be skipped (?) and mdl reports wrong line numbers.

### Environment

Using  `-s rules.rb`:
```ruby
all
exclude_rule 'MD007'
exclude_rule 'MD013'
exclude_rule 'MD029'
exclude_rule 'MD033'
rule 'MD009', :br_spaces => 2
```

mdl-0.13.0

### Expected Behavior

Correct line numbers expected.
### Actual Behavior

Line numbers get _shifted_ by the number of lines in the yaml code block (or in between the `---` document delimiters)
## Replication Case
Example minimum test case:
- (using above `-s rules.rb` )
- `Test.md`:
  ````markdown
  # A title
  # Another title without spacing to trigger an error
  ```yaml
  ---
  variable: "value"
  ---
  ``
  ````
- when executing `mdl -s rules.rb Test.md` produces:  
  ```console
  $ mdl -s rules.rb Test.md
  Test.md:1: MD022 Headers should be surrounded by blank lines
  Test.md:3: MD031 Fenced code blocks should be surrounded by blank lines
  ```
- and when executing with `mdl --ignore-front-matter -s rules.rb Test.md` produces:  
  ```console
  $ mdl --ignore-front-matter -s rules.rb Test.md
  Test.md:4: MD022 Headers should be surrounded by blank lines
  Test.md:6: MD031 Fenced code blocks should be surrounded by blank lines
  ```

The above example shows how the line numbers change depending on `--ignore-front-matter` being specified, in larger documents, this shifts all line number references by the number of lines inside the yaml code block.
